scm-core/src/main/java/sonia/scm/repository/work/SimpleWorkingCopyFactory.java

```diff
@@ -0,0 +1,311 @@
+ *                            │                               │                                 │                            <───┘                    │                        │
+ *                     ┌──────┴──────┐          ┌─────────────┴─────────────┐          ┌────────┴─────────┐          ┌───────┴───────┐          ┌─────┴─────┐          ┌───────┴───────┐
+ *                     │ModifyCommand│          │SimpleGitWorkingCopyFactory│          │WorkingCopyContext│          │WorkingCopyPool│          │WorkingCopy│          │WorkdirProvider│
+ *                     └─────────────┘          └───────────────────────────┘          └──────────────────┘          └───────────────┘          └───────────┘          └───────────────┘
+ * </pre>
+ * <img src="http://www.plantuml.com/plantuml/png/jLF1JiCm3BtdAtAkr7r0aQf9XN42JGE9SvHumyA9goHbAr-FnZgKDbCPGXnZl_ViFDlB49MFdINnm0QtVSFMAcVA-WbjIt2FyONz6xfTmss_KZgoxsKbjGSL8Kc96NnPooJOi8jmY4LHdKJccKbipKpL3bAOs7dkMldiUnbPUj2aq8e9fwppwjKPgoYUUJ9qMaC8suv4pXYf5CL5H2sdxQQz0WNuhhLLIE5cy54ws5yKF6I2cnD_fP30t1qqj17PNVwoGR_s_8u6_E3r8-o7X9W0odfgzLKseiE8Yl03_iSoP_8svbQpabVlP3rQ-35niLXCxo59LuQFhvzGcZYCR9azgW4-WxY2diJ_gBI1bWCUtx-xJtqQR7FKo6UNmvL-XLlqy2Kdbk1CP-aJ"/>
+ * @param <R> Type of central repository location
+ * @param <W> Type of working copy for repository
+ * @param <C> Type of repository context
+ */
+/*
+http://www.plantuml.com/plantuml/uml/jLF1JiCm3BtdAtAkr7r0aQf9XN42JGE9SvHumyA9goHbAr-FnZgKDbCPGXnZl_ViFDlB49MFdINnm0QtVSFMAcVA-WbjIt2FyONz6xfTmss_KZgoxsKbjGSL8Kc96NnPooJOi8jmY4LHdKJccKbipKpL3bAOs7dkMldiUnbPUj2aq8e9fwppwjKPgoYUUJ9qMaC8suv4pXYf5CL5H2sdxQQz0WNuhhLLIE5cy54ws5yKF6I2cnD_fP30t1qqj17PNVwoGR_s_8u6_E3r8-o7X9W0odfgzLKseiE8Yl03_iSoP_8svbQpabVlP3rQ-35niLXCxo59LuQFhvzGcZYCR9azgW4-WxY2diJ_gBI1bWCUtx-xJtqQR7FKo6UNmvL-XLlqy2Kdbk1CP-aJ
+@startuml
+ModifyCommand->SimpleGitWorkingCopyFactory : createWorkingCopy
+SimpleGitWorkingCopyFactory-> WorkingCopyContext**:create
+SimpleGitWorkingCopyFactory->WorkingCopyPool:getWorkingCopy
+group Try to reclaim
+WorkingCopyPool->WorkingCopyContext:reclaim
+alt reclaim successful
+WorkingCopyContext->WorkingCopy**
+WorkingCopyContext->> WorkingCopyPool:WorkingCopy
+else reclaim fails; create new
+WorkingCopyContext->x WorkingCopyPool:ReclaimFailedException
+WorkingCopyPool->WorkdirProvider:createNewWorkdir
+WorkdirProvider->>WorkingCopyPool
+WorkingCopyPool->WorkingCopyContext:initialize
+WorkingCopyContext->WorkingCopy**
+WorkingCopyContext->> WorkingCopyPool:WorkingCopy
+end
+WorkingCopyPool->>SimpleGitWorkingCopyFactory: WorkingCopy
+SimpleGitWorkingCopyFactory->>ModifyCommand: WorkingCopy
+...
+ModifyCommand->WorkingCopy:doWork
+...
+ModifyCommand->WorkingCopy:close
+WorkingCopy->SimpleGitWorkingCopyFactory:close
+SimpleGitWorkingCopyFactory->SimpleGitWorkingCopyFactory:closeWorkingCopy
+SimpleGitWorkingCopyFactory->SimpleGitWorkingCopyFactory:closeRepository
+SimpleGitWorkingCopyFactory->WorkingCopyPool:contextClosed
+WorkingCopyPool->WorkingCopyPool:cacheDirectory
+@enduml
+*/
```

Comment:
Do we really need ascii art, plantuml image and plantuml as text? I think we could remove at least the plantuml text, because it is already in the url.

Reply:
An image would definitely be better. This was just ment for the development process.